code/modules/fishing/fish/_fish.dm

```diff
@@ -168,6 +168,32 @@
 		progenitors = full_capitalize(name) //default value
 
 	register_evolutions()
+	register_item_context()
+
+/obj/item/fish/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	if(HAS_TRAIT(source, TRAIT_CATCH_AND_RELEASE))
+		context[SCREENTIP_CONTEXT_RMB] = "Release"
+		return CONTEXTUAL_SCREENTIP_SET
+	return NONE
+
+/obj/item/fish/interact_with_atom_secondary(atom/interacting_with, mob/living/user, list/modifiers)
+	if(!HAS_TRAIT(interacting_with, TRAIT_CATCH_AND_RELEASE))
+		return NONE
+	if(HAS_TRAIT(src, TRAIT_NODROP))
+		balloon_alert(user, "it's stuck to your hand!")
+		return ITEM_INTERACT_BLOCKING
+	balloon_alert(user, "releasing fish...")
+	if(!do_after(src, 3 SECONDS, interacting_with))
+		return ITEM_INTERACT_BLOCKING
+	balloon_alert(user, "fish released")
+	var/fish_dead_not_naive = status == FISH_DEAD && !HAS_MIND_TRAIT(user, TRAIT_NAIVE)
+	user.visible_message(span_notice("[user] releases [src] into [interacting_with]"), \
+		span_notice("You release [src] into [interacting_with]. [fish_dead_not_naive ? "May it rest in peace..." : "Bye bye [name]."]"), \
+		span_notice("You hear a splash."))
```

Comment:
`fish_dead_not_naive` is non ideal, my gut is making the goodbye text a var would work better here